Use more smart pointers in editing code

cdumez opened the pull request:
#### 3b05ce368eccb0c5c42dbdeb17e541e6418fbef8
<pre>
Use more smart pointers in editing code
<a href="https://bugs.webkit.org/show_bug.cgi?id=262200">https://bugs.webkit.org/show_bug.cgi?id=262200</a>

Reviewed by Brent Fulgham.

* Source/WebCore/dom/Position.cpp:
(WebCore::Position::protectedContainerText const):
* Source/WebCore/dom/Position.h:
* Source/WebCore/editing/AlternativeTextController.cpp:
(WebCore::AlternativeTextController::startAlternativeTextUITimer):
(WebCore::AlternativeTextController::stopPendingCorrection):
(WebCore::AlternativeTextController::isSpellingMarkerAllowed const):
(WebCore::AlternativeTextController::show):
(WebCore::AlternativeTextController::dismiss):
(WebCore::AlternativeTextController::dismissSoon):
(WebCore::AlternativeTextController::applyAutocorrectionBeforeTypingIfAppropriate):
(WebCore::AlternativeTextController::respondToUnappliedSpellCorrection):
(WebCore::AlternativeTextController::timerFired):
(WebCore::AlternativeTextController::handleAlternativeTextUIResult):
(WebCore::AlternativeTextController::canEnableAutomaticSpellingCorrection const):
(WebCore::AlternativeTextController::rootViewRectForRange const):
(WebCore::AlternativeTextController::respondToChangedSelection):
(WebCore::AlternativeTextController::editorClient):
(WebCore::AlternativeTextController::textChecker):
(WebCore::AlternativeTextController::recordAutocorrectionResponse):
(WebCore::AlternativeTextController::recordSpellcheckerResponseForModifiedCorrection):
(WebCore::AlternativeTextController::processMarkersOnTextToBeReplacedByResult):
(WebCore::AlternativeTextController::respondToMarkerAtEndOfWord):
(WebCore::AlternativeTextController::alternativeTextClient):
(WebCore::AlternativeTextController::applyAlternativeTextToRange):
(WebCore::AlternativeTextController::removeCorrectionIndicatorMarkers):
(WebCore::AlternativeTextController::insertDictatedText):
(WebCore::AlternativeTextController::removeDictationAlternativesForMarker):
(WebCore::AlternativeTextController::dictationAlternativesForMarker):
(WebCore::AlternativeTextController::applyDictationAlternative):
* Source/WebCore/editing/AlternativeTextController.h:
(WebCore::AlternativeTextController::protectedDocument const):
* Source/WebCore/editing/AppendNodeCommand.cpp:
(WebCore::AppendNodeCommand::doApply):
(WebCore::AppendNodeCommand::doUnapply):
* Source/WebCore/editing/AppendNodeCommand.h:
(WebCore::AppendNodeCommand::protectedParent const):
(WebCore::AppendNodeCommand::protectedNode const):
* Source/WebCore/editing/ApplyBlockElementCommand.cpp:
(WebCore::ApplyBlockElementCommand::doApply):
(WebCore::ApplyBlockElementCommand::formatSelection):
(WebCore::isNewLineAtPosition):
(WebCore::ApplyBlockElementCommand::renderStyleOfEnclosingTextNode):
(WebCore::ApplyBlockElementCommand::rangeForParagraphSplittingTextNodesIfNeeded):
(WebCore::ApplyBlockElementCommand::endOfNextParagraphSplittingTextNodesIfNeeded):
(WebCore::ApplyBlockElementCommand::createBlockElement):
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::applyBlockStyle):
(WebCore::ApplyStyleCommand::splitAncestorsWithUnicodeBidi):
(WebCore::ApplyStyleCommand::applyInlineStyle):
(WebCore::ApplyStyleCommand::applyInlineStyleToNodeRange):
(WebCore::ApplyStyleCommand::replaceWithSpanOrRemoveIfWithoutAttributes):
(WebCore::ApplyStyleCommand::highestAncestorWithConflictingInlineStyle):
(WebCore::ApplyStyleCommand::applyInlineStyleToPushDown):
(WebCore::ApplyStyleCommand::pushDownInlineStyleAroundNode):
(WebCore::ApplyStyleCommand::nodeFullySelected const):
(WebCore::ApplyStyleCommand::nodeFullyUnselected const):
(WebCore::ApplyStyleCommand::splitTextAtStart):
(WebCore::ApplyStyleCommand::splitTextAtEnd):
(WebCore::ApplyStyleCommand::splitTextElementAtStart):
(WebCore::ApplyStyleCommand::splitTextElementAtEnd):
(WebCore::ApplyStyleCommand::isValidCaretPositionInTextNode):
(WebCore::ApplyStyleCommand::mergeStartWithPreviousIfIdentical):
(WebCore::ApplyStyleCommand::mergeEndWithNextIfIdentical):
(WebCore::ApplyStyleCommand::joinChildTextNodes):
* Source/WebCore/editing/BreakBlockquoteCommand.cpp:
(WebCore::BreakBlockquoteCommand::doApply):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::EditCommandComposition::unapply):
(WebCore::EditCommandComposition::reapply):
(WebCore::CompositeEditCommand::willApplyCommand):
(WebCore::CompositeEditCommand::didApplyCommand):
(WebCore::CompositeEditCommand::composition const):
(WebCore::CompositeEditCommand::ensureComposition):
(WebCore::CompositeEditCommand::isRemovableBlock):
(WebCore::CompositeEditCommand::replaceElementWithSpanPreservingChildrenAndAttributes):
(WebCore::CompositeEditCommand::textNodeForRebalance const):
(WebCore::CompositeEditCommand::prepareWhitespaceAtPositionForSplit):
(WebCore::CompositeEditCommand::replaceCollapsibleWhitespaceWithNonBreakingSpaceIfNeeded):
(WebCore::CompositeEditCommand::deleteInsignificantText):
(WebCore::CompositeEditCommand::addBlockPlaceholderIfNeeded):
(WebCore::CompositeEditCommand::removePlaceholderAt):
(WebCore::CompositeEditCommand::moveParagraphContentsToNewBlockIfNecessary):
(WebCore::CompositeEditCommand::cloneParagraphUnderNewElement):
(WebCore::CompositeEditCommand::cleanupAfterDeletion):
(WebCore::CompositeEditCommand::moveParagraphWithClones):
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/CompositeEditCommand.h:
(WebCore::EditCommandComposition::protectedDocument const):
* Source/WebCore/editing/CreateLinkCommand.cpp:
(WebCore::CreateLinkCommand::doApply):
* Source/WebCore/editing/CustomUndoStep.cpp:
(WebCore::CustomUndoStep::didRemoveFromUndoManager):
* Source/WebCore/editing/DeleteFromTextNodeCommand.cpp:
(WebCore::DeleteFromTextNodeCommand::doApply):
(WebCore::DeleteFromTextNodeCommand::doUnapply):
(WebCore::DeleteFromTextNodeCommand::getNodesInCommand):
(WebCore::DeleteFromTextNodeCommand::protectedNode const):
* Source/WebCore/editing/DeleteFromTextNodeCommand.h:
* Source/WebCore/editing/DeleteSelectionCommand.cpp:
(WebCore::isSpecialHTMLElement):
(WebCore::positionBeforeContainingSpecialElement):
(WebCore::positionAfterContainingSpecialElement):
(WebCore::DeleteSelectionCommand::saveTypingStyleState):
(WebCore::DeleteSelectionCommand::insertBlockPlaceholderForTableCellIfNeeded):
(WebCore::DeleteSelectionCommand::removeNodeUpdatingStates):
(WebCore::DeleteSelectionCommand::removeNode):
(WebCore::DeleteSelectionCommand::handleGeneralDelete):
(WebCore::DeleteSelectionCommand::fixupWhitespace):
(WebCore::DeleteSelectionCommand::mergeParagraphs):
(WebCore::DeleteSelectionCommand::removePreviouslySelectedEmptyTableRows):
(WebCore::DeleteSelectionCommand::calculateTypingStyleAfterDelete):
(WebCore::DeleteSelectionCommand::originalStringForAutocorrectionAtBeginningOfSelection):
(WebCore::DeleteSelectionCommand::doApply):
* Source/WebCore/editing/DeleteSelectionCommand.h:
(WebCore::DeleteSelectionCommand::protectedStartBlock const):
(WebCore::DeleteSelectionCommand::protectedEndBlock const):
(WebCore::DeleteSelectionCommand::protectedEndTableRow const):
* Source/WebCore/editing/DictationCommand.cpp:
(WebCore::DictationCommandLineOperation::DictationCommandLineOperation):
(WebCore::DictationCommandLineOperation::operator() const):
(WebCore::DictationMarkerSupplier::create):
(WebCore::DictationMarkerSupplier::DictationMarkerSupplier):
(WebCore::DictationCommand::insertText):
(WebCore::DictationCommand::doApply):
(WebCore::DictationCommand::insertTextRunWithoutNewlines):
* Source/WebCore/editing/DictationCommand.h:
* Source/WebCore/editing/EditCommand.cpp:
(WebCore::EditCommand::setParent):
(WebCore::EditCommand::postTextStateChangeNotification):
(WebCore::SimpleEditCommand::addNodeAndDescendants):
* Source/WebCore/editing/EditCommand.h:
* Source/WebCore/editing/Editing.cpp:
(WebCore::isEditablePosition):
(WebCore::isRichlyEditablePosition):
(WebCore::editableRootForPosition):
(WebCore::nextVisuallyDistinctCandidate):
(WebCore::previousVisuallyDistinctCandidate):
(WebCore::firstEditablePositionAfterPositionInRoot):
(WebCore::lastEditablePositionBeforePositionInRoot):
(WebCore::isFirstPositionAfterTable):
(WebCore::isLastPositionBeforeTable):
(WebCore::closestEditablePositionInElementForAbsolutePoint):
(WebCore::highestEnclosingNodeOfType):
(WebCore::hasARenderedDescendant):
(WebCore::highestNodeToRemoveInPruning):
(WebCore::enclosingAnchorElement):
(WebCore::enclosingList):
(WebCore::enclosingListChild):
(WebCore::enclosingEmptyListItem):
(WebCore::outermostEnclosingList):
(WebCore::canMergeLists):
(WebCore::nextNodeConsideringAtomicNodes):
(WebCore::isEmptyTableCell):
(WebCore::selectionForParagraphIteration):
(WebCore::indexForVisiblePosition):
(WebCore::localCaretRectInRendererForRect):
(WebCore::visibleImageElementsInRangeWithNonLoadedImages):
* Source/WebCore/editing/Editing.h:
* Source/WebCore/editing/InsertListCommand.cpp:
(WebCore::enclosingListChild):
(WebCore::InsertListCommand::selectionHasListOfType):
(WebCore::InsertListCommand::doApplyForSingleParagraph):
(WebCore::adjacentEnclosingList):
* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::deleteKeyPressed):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::VisibleSelection::adjustSelectionRespectingGranularity):
* Source/WebCore/page/AlternativeTextClient.h:
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::startDrag):
* Source/WebCore/page/EditorClient.h:

Canonical link: <a href="https://commits.webkit.org/268548@main">https://commits.webkit.org/268548@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/254ec02d3ce7da2002166b8d5b1bfaef60259001

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19984 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20411 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21032 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21880 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18666 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20216 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23662 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20560 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20167 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20204 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20155 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17377 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22731 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17325 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18173 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24434 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18402 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18349 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22423 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18939 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16064 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18120 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4791 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22469 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18763 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/editing/ApplyStyleCommand.cpp:
```diff
@@ -988,7 +992,7 @@ RefPtr<HTMLElement> ApplyStyleCommand::highestAncestorWithConflictingInlineStyle
 
     for (RefPtr ancestor = node; ancestor; ancestor = ancestor->parentNode()) {
         if (is<HTMLElement>(*ancestor) && shouldRemoveInlineStyleFromElement(style, downcast<HTMLElement>(*ancestor)))
```
Maybe this should be written in terms of `dynamicDowncast`, instead of checking `is<HTMLElement>` then down casting multiple times?

brentfulgham commented on Source/WebCore/editing/BreakBlockquoteCommand.cpp:
```diff
@@ -128,17 +128,17 @@ void BreakBlockquoteCommand::doApply()
-            if (auto* nextNode = NodeTraversal::next(*startNode))
-                startNode = nextNode;
+        Ref textNode = downcast<Text>(*startNode);
+        if ((unsigned)pos.deprecatedEditingOffset() >= textNode->length()) {
```
We should use a C++ cast for unsigned. Not sure if it's just a `static_cast<unsigned>` in this case.

brentfulgham commented on Source/WebCore/editing/DictationCommand.cpp:
```diff
@@ -40,31 +40,32 @@ namespace WebCore {
 class DictationMarkerSupplier : public TextInsertionMarkerSupplier {
 public:
-    static Ref<DictationMarkerSupplier> create(const Vector<DictationAlternative>& alternatives)
+    static Ref<DictationMarkerSupplier> create(Vector<DictationAlternative>&& alternatives)
```
That's nice! Being able to move that vector in.

brentfulgham reviewed (approved):
Looks good. I had a few minor suggestions. r=me

cdumez committed:
Use more smart pointers in editing code
https://bugs.webkit.org/show_bug.cgi?id=262200

Reviewed by Brent Fulgham.

* Source/WebCore/dom/Position.cpp:
(WebCore::Position::protectedContainerText const):
* Source/WebCore/dom/Position.h:
* Source/WebCore/editing/AlternativeTextController.cpp:
(WebCore::AlternativeTextController::startAlternativeTextUITimer):
(WebCore::AlternativeTextController::stopPendingCorrection):
(WebCore::AlternativeTextController::isSpellingMarkerAllowed const):
(WebCore::AlternativeTextController::show):
(WebCore::AlternativeTextController::dismiss):
(WebCore::AlternativeTextController::dismissSoon):
(WebCore::AlternativeTextController::applyAutocorrectionBeforeTypingIfAppropriate):
(WebCore::AlternativeTextController::respondToUnappliedSpellCorrection):
(WebCore::AlternativeTextController::timerFired):
(WebCore::AlternativeTextController::handleAlternativeTextUIResult):
(WebCore::AlternativeTextController::canEnableAutomaticSpellingCorrection const):
(WebCore::AlternativeTextController::rootViewRectForRange const):
(WebCore::AlternativeTextController::respondToChangedSelection):
(WebCore::AlternativeTextController::editorClient):
(WebCore::AlternativeTextController::textChecker):
(WebCore::AlternativeTextController::recordAutocorrectionResponse):
(WebCore::AlternativeTextController::recordSpellcheckerResponseForModifiedCorrection):
(WebCore::AlternativeTextController::processMarkersOnTextToBeReplacedByResult):
(WebCore::AlternativeTextController::respondToMarkerAtEndOfWord):
(WebCore::AlternativeTextController::alternativeTextClient):
(WebCore::AlternativeTextController::applyAlternativeTextToRange):
(WebCore::AlternativeTextController::removeCorrectionIndicatorMarkers):
(WebCore::AlternativeTextController::insertDictatedText):
(WebCore::AlternativeTextController::removeDictationAlternativesForMarker):
(WebCore::AlternativeTextController::dictationAlternativesForMarker):
(WebCore::AlternativeTextController::applyDictationAlternative):
* Source/WebCore/editing/AlternativeTextController.h:
(WebCore::AlternativeTextController::protectedDocument const):
* Source/WebCore/editing/AppendNodeCommand.cpp:
(WebCore::AppendNodeCommand::doApply):
(WebCore::AppendNodeCommand::doUnapply):
* Source/WebCore/editing/AppendNodeCommand.h:
(WebCore::AppendNodeCommand::protectedParent const):
(WebCore::AppendNodeCommand::protectedNode const):
* Source/WebCore/editing/ApplyBlockElementCommand.cpp:
(WebCore::ApplyBlockElementCommand::doApply):
(WebCore::ApplyBlockElementCommand::formatSelection):
(WebCore::isNewLineAtPosition):
(WebCore::ApplyBlockElementCommand::renderStyleOfEnclosingTextNode):
(WebCore::ApplyBlockElementCommand::rangeForParagraphSplittingTextNodesIfNeeded):
(WebCore::ApplyBlockElementCommand::endOfNextParagraphSplittingTextNodesIfNeeded):
(WebCore::ApplyBlockElementCommand::createBlockElement):
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::applyBlockStyle):
(WebCore::ApplyStyleCommand::splitAncestorsWithUnicodeBidi):
(WebCore::ApplyStyleCommand::applyInlineStyle):
(WebCore::ApplyStyleCommand::applyInlineStyleToNodeRange):
(WebCore::ApplyStyleCommand::replaceWithSpanOrRemoveIfWithoutAttributes):
(WebCore::ApplyStyleCommand::highestAncestorWithConflictingInlineStyle):
(WebCore::ApplyStyleCommand::applyInlineStyleToPushDown):
(WebCore::ApplyStyleCommand::pushDownInlineStyleAroundNode):
(WebCore::ApplyStyleCommand::nodeFullySelected const):
(WebCore::ApplyStyleCommand::nodeFullyUnselected const):
(WebCore::ApplyStyleCommand::splitTextAtStart):
(WebCore::ApplyStyleCommand::splitTextAtEnd):
(WebCore::ApplyStyleCommand::splitTextElementAtStart):
(WebCore::ApplyStyleCommand::splitTextElementAtEnd):
(WebCore::ApplyStyleCommand::isValidCaretPositionInTextNode):
(WebCore::ApplyStyleCommand::mergeStartWithPreviousIfIdentical):
(WebCore::ApplyStyleCommand::mergeEndWithNextIfIdentical):
(WebCore::ApplyStyleCommand::joinChildTextNodes):
* Source/WebCore/editing/BreakBlockquoteCommand.cpp:
(WebCore::BreakBlockquoteCommand::doApply):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::EditCommandComposition::unapply):
(WebCore::EditCommandComposition::reapply):
(WebCore::CompositeEditCommand::willApplyCommand):
(WebCore::CompositeEditCommand::didApplyCommand):
(WebCore::CompositeEditCommand::composition const):
(WebCore::CompositeEditCommand::ensureComposition):
(WebCore::CompositeEditCommand::isRemovableBlock):
(WebCore::CompositeEditCommand::replaceElementWithSpanPreservingChildrenAndAttributes):
(WebCore::CompositeEditCommand::textNodeForRebalance const):
(WebCore::CompositeEditCommand::prepareWhitespaceAtPositionForSplit):
(WebCore::CompositeEditCommand::replaceCollapsibleWhitespaceWithNonBreakingSpaceIfNeeded):
(WebCore::CompositeEditCommand::deleteInsignificantText):
(WebCore::CompositeEditCommand::addBlockPlaceholderIfNeeded):
(WebCore::CompositeEditCommand::removePlaceholderAt):
(WebCore::CompositeEditCommand::moveParagraphContentsToNewBlockIfNecessary):
(WebCore::CompositeEditCommand::cloneParagraphUnderNewElement):
(WebCore::CompositeEditCommand::cleanupAfterDeletion):
(WebCore::CompositeEditCommand::moveParagraphWithClones):
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/CompositeEditCommand.h:
(WebCore::EditCommandComposition::protectedDocument const):
* Source/WebCore/editing/CreateLinkCommand.cpp:
(WebCore::CreateLinkCommand::doApply):
* Source/WebCore/editing/CustomUndoStep.cpp:
(WebCore::CustomUndoStep::didRemoveFromUndoManager):
* Source/WebCore/editing/DeleteFromTextNodeCommand.cpp:
(WebCore::DeleteFromTextNodeCommand::doApply):
(WebCore::DeleteFromTextNodeCommand::doUnapply):
(WebCore::DeleteFromTextNodeCommand::getNodesInCommand):
(WebCore::DeleteFromTextNodeCommand::protectedNode const):
* Source/WebCore/editing/DeleteFromTextNodeCommand.h:
* Source/WebCore/editing/DeleteSelectionCommand.cpp:
(WebCore::isSpecialHTMLElement):
(WebCore::positionBeforeContainingSpecialElement):
(WebCore::positionAfterContainingSpecialElement):
(WebCore::DeleteSelectionCommand::saveTypingStyleState):
(WebCore::DeleteSelectionCommand::insertBlockPlaceholderForTableCellIfNeeded):
(WebCore::DeleteSelectionCommand::removeNodeUpdatingStates):
(WebCore::DeleteSelectionCommand::removeNode):
(WebCore::DeleteSelectionCommand::handleGeneralDelete):
(WebCore::DeleteSelectionCommand::fixupWhitespace):
(WebCore::DeleteSelectionCommand::mergeParagraphs):
(WebCore::DeleteSelectionCommand::removePreviouslySelectedEmptyTableRows):
(WebCore::DeleteSelectionCommand::calculateTypingStyleAfterDelete):
(WebCore::DeleteSelectionCommand::originalStringForAutocorrectionAtBeginningOfSelection):
(WebCore::DeleteSelectionCommand::doApply):
* Source/WebCore/editing/DeleteSelectionCommand.h:
(WebCore::DeleteSelectionCommand::protectedStartBlock const):
(WebCore::DeleteSelectionCommand::protectedEndBlock const):
(WebCore::DeleteSelectionCommand::protectedEndTableRow const):
* Source/WebCore/editing/DictationCommand.cpp:
(WebCore::DictationCommandLineOperation::DictationCommandLineOperation):
(WebCore::DictationCommandLineOperation::operator() const):
(WebCore::DictationMarkerSupplier::create):
(WebCore::DictationMarkerSupplier::DictationMarkerSupplier):
(WebCore::DictationCommand::insertText):
(WebCore::DictationCommand::doApply):
(WebCore::DictationCommand::insertTextRunWithoutNewlines):
* Source/WebCore/editing/DictationCommand.h:
* Source/WebCore/editing/EditCommand.cpp:
(WebCore::EditCommand::setParent):
(WebCore::EditCommand::postTextStateChangeNotification):
(WebCore::SimpleEditCommand::addNodeAndDescendants):
* Source/WebCore/editing/EditCommand.h:
* Source/WebCore/editing/Editing.cpp:
(WebCore::isEditablePosition):
(WebCore::isRichlyEditablePosition):
(WebCore::editableRootForPosition):
(WebCore::nextVisuallyDistinctCandidate):
(WebCore::previousVisuallyDistinctCandidate):
(WebCore::firstEditablePositionAfterPositionInRoot):
(WebCore::lastEditablePositionBeforePositionInRoot):
(WebCore::isFirstPositionAfterTable):
(WebCore::isLastPositionBeforeTable):
(WebCore::closestEditablePositionInElementForAbsolutePoint):
(WebCore::highestEnclosingNodeOfType):
(WebCore::hasARenderedDescendant):
(WebCore::highestNodeToRemoveInPruning):
(WebCore::enclosingAnchorElement):
(WebCore::enclosingList):
(WebCore::enclosingListChild):
(WebCore::enclosingEmptyListItem):
(WebCore::outermostEnclosingList):
(WebCore::canMergeLists):
(WebCore::nextNodeConsideringAtomicNodes):
(WebCore::isEmptyTableCell):
(WebCore::selectionForParagraphIteration):
(WebCore::indexForVisiblePosition):
(WebCore::localCaretRectInRendererForRect):
(WebCore::visibleImageElementsInRangeWithNonLoadedImages):
* Source/WebCore/editing/Editing.h:
* Source/WebCore/editing/InsertListCommand.cpp:
(WebCore::enclosingListChild):
(WebCore::InsertListCommand::selectionHasListOfType):
(WebCore::InsertListCommand::doApplyForSingleParagraph):
(WebCore::adjacentEnclosingList):
* Source/WebCore/editing/TypingCommand.cpp:
(WebCore::TypingCommand::deleteKeyPressed):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::VisibleSelection::adjustSelectionRespectingGranularity):
* Source/WebCore/page/AlternativeTextClient.h:
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::startDrag):
* Source/WebCore/page/EditorClient.h:

Canonical link: https://commits.webkit.org/268548@main
